android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/PayloadHelper.java

```diff
@@ -65,9 +65,13 @@ private static Uri checkRequirements(String uriString) {
       return null;
     }
     Uri uri = Uri.parse(uriString);
-    if (!uri.getScheme()
-        .equalsIgnoreCase(SCHEME)) {
-      throw new IllegalArgumentException();
+    try {
+      if (!uri.getScheme().equalsIgnoreCase(SCHEME)) {
+        throw new IllegalArgumentException();
+      }
+    } catch (NullPointerException e) {
```

Comment:
instead pf using a try catch why don't you do the null checks first?

Reply:
i though that since its `getScheme` that may raise a NPE, it was easier this way than store `getScheme` result in a var an then null check it

Reply:
with a try catch you're hiding possible problems or outputting spam for no reason